用 Pandoc 生成定理环境

XiangyunHuang opened the pull request:
- [ ] 定制 css 定理类样式
- [x] 定制 LaTeX 定理类环境
- [ ] Travis 编译通过
- [ ] 合并进 master 分支

Commit:
create thm environments by pandoc+css

Commit:
add name attribute to theorems

XiangyunHuang commented:
> > 可以从 devel 分支向 master 分支发起 PR
> 
> github 我还不熟悉。PR=Pull Request?:)

不用担心，我已经越俎代庖了

Commit:
using lua to create latex thm environments for elegantbook.

lief-y commented:
> > > 可以从 devel 分支向 master 分支发起 PR
> > 
> > 
> > github 我还不熟悉。PR=Pull Request?:)
> 
> 不用担心，我已经越俎代庖了

太感谢了！
pandoc fenced_div 到LaTeX 定理环境也解决了。学习新东西真是不宜啊！添加了lua代码。去掉了原来bookdown.post.latex 相应的部分。现在应该html和pdf一致了。目前未考虑的一个问题是html的中定理的引用。暂时不知道咋搞。

lief-y commented:
@XiangyunHuang  发现了一个问题。定制的css无法随着章节的变化改变数字，因为每章是一个单独的html。好像可以加个js弄过去。不过还没有搞出来。这周暂时没时间了。可能要到周末才有空。

XiangyunHuang commented:
@fyemath  不急，慢慢来！不知道 @yihui 有没有想法，能不能给我们提供一下思路？后续我们可以把解决办法以 PR 的方式提交给 bookdown

Commit:
fixed theorem number issue using js and jquery

get chapter number using js and pass it to a variable in css for the chapter counter.

lief-y commented:
> @fyemath 不急，慢慢来！不知道 @yihui 有没有想法，能不能给我们提供一下思路？后续我们可以把解决办法以 PR 的方式提交给 bookdown

今天捣鼓了一天，目前暂时满意了。js部分其实很简单。因为bookdown会生成chapter number， 存在 numer attribute里面，只要用js把这个数字取出来，再传回到css就好了。

css里面虽然可以定义变量，但是不方面把attr 赋值过去。所以不得已用了css。不知道@yihui 有没有好的办法。

Commit:
重排了css，定理类名称使用变量，容易中英文切换。

添加了更多类的定理环境名称。引用的问题无法完美解决。可以额外添加一个id。但是不知道咋显示数字。

yihui commented:
这个看起来太厉害了，但我目前恐怕没空细琢磨，感觉大方向应该是对的。谢谢！

lief-y commented:
> 这个看起来太厉害了，但我目前恐怕没空细琢磨，感觉大方向应该是对的。谢谢！

可惜定理环境的引用有问题，加id比较容易，引用的时候显示定理的编号或者题目不知道咋实现。
